Junit/src/IPL.java

```diff
@@ -0,0 +1,364 @@
+        char[] maxRunScorerName = new char[maxNameSize];
+
+        for (Player player : listOfPlayers) {
+            if(player.getTeam().equals(team)){
+
+                if(higestWicketByPlayer < player.getWicket()){
+                    higestWicketByPlayer = player.getWicket();
+                    maxWicketPlayerName = player.getName().toCharArray();
+                }
+
+                if(highestRunScoredByPerson < player.getRuns()){
+                    highestRunScoredByPerson = player.getRuns();
+                    maxRunScorerName = player.getName().toCharArray();
+                }
+            }
+        }
+        System.out.println("Highest wicket taker is " +  String.valueOf(maxWicketPlayerName));
+        System.out.print("Highest run scorer is " + String.valueOf(maxRunScorerName));
+    }
+
+    // finding top three batsmen bowler and all rounder
+    // all rounder is found using a cumulative score of runs and wickets.
+    public static void findTopThree(List<Player> listOfPlayers){
+
+        List<Player> battingList = listOfPlayers.stream().filter(p -> p.getRole().equals("BATSMAN")).collect(Collectors.toList());
+        Collections.sort(battingList,new RunComparator());
+
+        System.out.println("===================== Batsman ================");
+
+        for(int i=0;i<3;i++){
+            System.out.print(battingList.get(i).getName() + " " + battingList.get(i).getRuns() + "\n");
+        }
+
+        System.out.println("===================== Bowler ================");
+
+        List<Player> bowlingList = listOfPlayers.stream().filter(p -> p.getRole().equals("BOWLER")).collect(Collectors.toList());
+        Collections.sort(bowlingList,new WicketComparator());
+
+        for(int i=0;i<3;i++){
+            System.out.print(bowlingList.get(i).getName() + " " + bowlingList.get(i).getWicket() + "\n");
+        }
+
+        System.out.println("============= All Rounders ==================");
+
+
+        List<Player> allRounderList = listOfPlayers.stream().filter(p -> p.getRole().equals("ALL ROUNDER")).collect(Collectors.toList());
+        Collections.sort(allRounderList,new RunWicketComparator());
+
+        for(int i=0;i<3;i++){
+            System.out.println(allRounderList.get(i).getName() + " " + allRounderList.get(i).cumulativeScore());
+        }
+    }
+
+    // if all team used best batsmen then we have to find winner team
+
+    /*
+        idea:
+            1) took 11 best batsmen from all team
+            2) add all runs scored by them
+            3) highest runs scored will be the winner
+
+    */
+
+    public String findBestBatsmenTeam(List<Player> listOfPlayers){
+        int totalPlayerInATeam = 11;
+        int maxTeamNameSize = 20;
+        int totalTeams = 8;
+        int maxRunsByTeam = 0;
+
+        char[] winnerTeam = new char[maxTeamNameSize];
+
+        String[] listOfTeams = new String[totalTeams];
+
+
+        listOfTeams[0] =  "CSK";
```

Comment:
Do not hard code values like this.
Could have been implemented better if the problem was modeled differently.